weasel/_util.py

```diff
@@ -171,23 +175,53 @@ def substitute_project_variables(
     return dict(interpolated["project"])
 
 
-def validate_project_version(config: Dict[str, Any]) -> None:
-    """If the project defines a compatible spaCy version range, chec that it's
-    compatible with the current version of spaCy.
+def validate_weasel_version(config: Dict[str, Any]) -> None:
+    """If the project defines a compatible Weasel version range, check that it's
+    compatible with the current version of Weasel.
 
     config (Dict[str, Any]): The loaded config.
     """
-    spacy_version = config.get("spacy_version", None)
-    if spacy_version and not is_compatible_version(about.__version__, spacy_version):
+    version = config.get("weasel_version", None)
+    if version and not is_compatible_version(about.__version__, version):
         err = (
-            f"The {PROJECT_FILE} specifies a spaCy version range ({spacy_version}) "
-            f"that's not compatible with the version of spaCy you're running "
+            f"The {PROJECT_FILE} specifies a Weasel version range ({version}) "
+            f"that's not compatible with the version of Weasel you're running "
             f"({about.__version__}). You can edit version requirement in the "
             f"{PROJECT_FILE} to load it, but the project may not run as expected."
         )
         msg.fail(err, exits=1)
 
 
+def validate_spacy_version(config: Dict[str, Any]) -> None:
+    """If the project defines a compatible spaCy version range, check that it's
+    compatible with the current version of spaCy.
+
+    config (Dict[str, Any]): The loaded config.
+    """
+    version = config.get("spacy_version", None)
+    if version:
+        try:
+            import spacy
+        except ImportError:
+            err = (
+                f"The {PROJECT_FILE} specifies a spaCy version range ({version}) "
+                f"but spaCy is not installed within the environment. "
+                f"Did you forget to add 'spacy{version}` to your requirement file?"
+            )
+            msg.fail(err, exits=1)
+
+        spacy_version = spacy.about.__version__
```

Comment:
I so desperately want to drop python 3.6-3.7, hmm. I'd much rather use `importlib.metadata` for this.

Reply:
We can use [`importlib_metadata`](https://importlib-metadata.readthedocs.io/en/latest/index.html)

Reply:
You have not heard my rant about `importlib_metadata`. We do have working solutions like using the vendored version we have in `catalogue`, but an absolute no to having `importlib_metadata` as a requirement.

Reply:
That's fine by me :)

Reply:
Ok I used `catalogue`'s utility !